client/src/details/UrlBuilder.cpp

```diff
@@ -133,15 +133,15 @@ UrlBuilder& UrlBuilder::ResetPath()
 UrlBuilder& UrlBuilder::SetQuery(const std::string& key, const std::string& value)
 {
     THROW_CODE_IF_LOG(InvalidArg, key.empty() || value.empty(), m_handler, "Query key and value must not empty");
-    const std::string query = key + "=" + value;
+    const std::string query = EscapeString(key) + "=" + EscapeString(value);
     THROW_IF_CURL_URL_SETUP_ERROR(curl_url_set(m_handle, CURLUPART_QUERY, query.c_str(), 0 /*flags*/));
     return *this;
 }
 
 UrlBuilder& UrlBuilder::AppendQuery(const std::string& key, const std::string& value)
 {
     THROW_CODE_IF_LOG(InvalidArg, key.empty() || value.empty(), m_handler, "Query key and value must not empty");
-    const std::string query = key + "=" + value;
+    const std::string query = EscapeString(key) + "=" + EscapeString(value);

```

Comment:
Rename EscapeString() to URLEncode() [*].  curl might have named their API easy_escape, but there's a bunch of escaping mechanisms and it would be good to be clearer about which this is using.

[*] As I mentioned previously, the RFC calls this "percent encoding", but I think "UrlEncode" is generally recognizable.